test/Feature/ByteAddressBuffer/ByteAddressBuffers.test

```diff
@@ -1,41 +1,98 @@
-// various `Load*` and `Store` methods on `[RW]ByteAddressBuffer`.
+// various `Load` and `Store` methods on `[RW]ByteAddressBuffer`.
 
-// The expected behaviour is to load the values in `In1` and `In2` at the given
+// The expected behaviour is to load the values in `In0` and `In1` at the given
 // byte-offset, add them, and store the result at the respective offset in
 // `Out`. We expect each load and store to only access mapped resource data, so
 // `CheckAccessFullyMapped` should always return `true = 1`.
 
-ByteAddressBuffer In1 : register(t0);
-ByteAddressBuffer In2 : register(t1);
+struct S0 {
+  int a;
+  uint b;
+};
+
+struct S1 {
+  int4 a;
+  int4 b;
+  uint4 c;
+  uint4 d;
+};
+
+ByteAddressBuffer In0 : register(t0);
+ByteAddressBuffer In1 : register(t1);
 RWByteAddressBuffer Out : register(u0);
-RWBuffer<uint> MappedBuf : register(u0, space1);
+RWBuffer<uint> Mapped : register(u0, space1);
 
 [numthreads(4,1,1)]
 void main() {
   uint status;
 
-  uint u1 = In1.Load(0, status);
-  MappedBuf[0] = CheckAccessFullyMapped(status);
-  uint v1 = In2.Load(0);
-  Out.Store(0, u1 + v1);
-
-  uint2 u2 = In1.Load2(16, status);
-  MappedBuf[1] = CheckAccessFullyMapped(status);
-  uint2 v2 = In2.Load2(16);
-  Out.Store2(16, u2 + v2);
-
-  uint3 u3 = In1.Load3(32, status);
-  MappedBuf[2] = CheckAccessFullyMapped(status);
-  uint3 v3 = In2.Load3(32);
-  Out.Store3(32, u3 + v3);
-
-  uint4 u4 = In1.Load4(48, status);
-  MappedBuf[3] = CheckAccessFullyMapped(status);
-  uint4 v4 = In2.Load4(48);
-  Out.Store4(48, u4 + v4);
+  // uint
+  uint U0 = In0.Load(0, status);
+  Mapped[0] = CheckAccessFullyMapped(status);
+  uint V0 = In1.Load(0);
+  Out.Store(0, U0 + V0);
+
+  uint2 U1 = In0.Load2(16, status);
+  Mapped[1] = CheckAccessFullyMapped(status);
+  uint2 V1 = In1.Load2(16);
+  Out.Store2(16, U1 + V1);
+
+  uint3 U2 = In0.Load3(32, status);
+  Mapped[2] = CheckAccessFullyMapped(status);
+  uint3 V2 = In1.Load3(32);
+  Out.Store3(32, U2 + V2);
+
+  uint4 U3 = In0.Load4(48, status);
+  Mapped[3] = CheckAccessFullyMapped(status);
+  uint4 V3 = In1.Load4(48);
+  Out.Store4(48, U3 + V3);
+
+  // bool
+  bool U4 = In0.Load<bool>(64, status);
+  Mapped[4] = CheckAccessFullyMapped(status);
+  bool V4 = In1.Load<bool>(64);
+  Out.Store<bool>(64, U4 + V4);
+
+  bool2 U5 = In0.Load<bool2>(64, status);
+  Mapped[5] = CheckAccessFullyMapped(status);
+  bool2 V5 = In1.Load<bool2>(64);
+  Out.Store<bool2>(80, U5 + V5);
+
+  bool3 U6 = In0.Load<bool3>(64, status);
+  Mapped[6] = CheckAccessFullyMapped(status);
+  bool3 V6 = In1.Load<bool3>(64);
+  Out.Store<bool3>(96, U6 + V6);
+
+  bool4 U7 = In0.Load<bool4>(64, status);
+  Mapped[7] = CheckAccessFullyMapped(status);
+  bool4 V7 = In1.Load<bool4>(64);
+  Out.Store<bool4>(112, U7 + V7);
+
+  // array
+  int U8[4] = In0.Load<int[4]>(0, status);
+  Mapped[8] = CheckAccessFullyMapped(status);
+  int V8[4] = In1.Load<int[4]>(0);
+  int TempArray[4];
+  for(int i = 0; i < 4; i++) {
```

Comment:
nit: Should we follow coding standards in test code and make this a capital 'I' ?